lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/searcher/RemoteCentralRepositorySearcher.java

```diff
@@ -68,74 +99,95 @@ public Collection<Artifact> getArtifacts(Dependency artifactToSearch) {
+				});
+				
+				return artifactInfos;
+			}
 		});
-		return artifactInfos;
 	}
-	
-	private Set<ArtifactVersion> internalGetArtifactVersions(Dependency artifactToSearch, String packaging) {
+
+	private static CacheManager<Request, Collection<ArtifactVersion>> artifactVersionsCache = new CacheManager<>(cacheExpirationTimeoutMinutes);
+	private Collection<ArtifactVersion> internalGetArtifactVersions(Dependency artifactToSearch, String packaging) {
 		if (isEmpty(artifactToSearch.getArtifactId())) {
 			return Collections.emptySet();
 		}
 
 		Request request = createArtifactVersionsRequest(artifactToSearch, packaging);
-		JsonObject responseBody = getResponseBody(request, artifactToSearch);
-		if (responseBody == null || responseBody.get("numFound").getAsInt() <= 0 ) {
-			return Collections.emptySet();
-		}
-
-		Set<ArtifactVersion> artifactVersions = new HashSet<ArtifactVersion>();
-		responseBody.get("docs").getAsJsonArray().forEach(d -> {
-			artifactVersions.add(new DefaultArtifactVersion(d.getAsJsonObject().get(VERSION).getAsString()));
+		return artifactVersionsCache.get(request, new Callable<Collection<ArtifactVersion>>() {
+			@Override
+			public Collection<ArtifactVersion> call() throws Exception {
+				JsonObject responseBody = getResponseBody(request, artifactToSearch);
+				if (responseBody == null || responseBody.get("numFound").getAsInt() <= 0 ) {
+					return Collections.emptySet();
+				}
+	
+				Set<ArtifactVersion> artifactVersions = new HashSet<ArtifactVersion>();
+				responseBody.get("docs").getAsJsonArray().forEach(d -> {
+					artifactVersions.add(new DefaultArtifactVersion(d.getAsJsonObject().get(VERSION).getAsString()));
+				});
+	
+				return artifactVersions;
+			}
 		});
-		return artifactVersions;
 	}
 	
-	private Set<String> internalGetGroupIds(Dependency artifactToSearch, String packaging) {
+	private static CacheManager<Request, Collection<String>> groupIdsCache = new CacheManager<>(cacheExpirationTimeoutMinutes);
+	private Collection<String> internalGetGroupIds(Dependency artifactToSearch, String packaging) {
 		if (isEmpty(artifactToSearch.getGroupId())) {
 			return Collections.emptySet();
 		}
 		
 		Request request = createGroupIdsRequest(artifactToSearch, packaging);
-		JsonObject responseBody = getResponseBody(request, artifactToSearch);
-		if (responseBody == null || responseBody.get("numFound").getAsInt() <= 0 ) {
-			return Collections.emptySet();
-		}
-		
-		Set<String> artifactGroupIds = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
-		responseBody.get("docs").getAsJsonArray().forEach(d -> {
-			artifactGroupIds.add(d.getAsJsonObject().get(GROUP_ID).getAsString());
+		return groupIdsCache.get(request, new Callable<Collection<String>>() {
+			@Override
+			public Collection<String> call() throws Exception {
+				JsonObject responseBody = getResponseBody(request, artifactToSearch);
+				if (responseBody == null || responseBody.get("numFound").getAsInt() <= 0 ) {
```

Comment:
same here with "numFound"

I suggest too that you add a sample of JSON in the javadoc of the head JSONMavenRemoteCentralConstants class

Reply:
See previous reply for `docs` and `numFound` values

Reply:
I've added [JsonRemoteCentralRepositoryConstants](https://github.com/eclipse/lemminx-maven/blob/df3ee17bfa87af44ff70799c19beec44069ad2c0/lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/searcher/JsonRemoteCentralRepositoryConstants.java) class with JavaDoc comments and an example